test/parallel/test-runner-run-coverage.mjs

```diff
@@ -0,0 +1,182 @@
+          });
+          assert.throws(() => run({ coverage: true, functionCoverage: [functionCoverage] }), {
+            code: 'ERR_INVALID_ARG_TYPE'
+          });
+        });
+
+      assert.throws(() => run({ coverage: true, functionCoverage: -1 }), { code: 'ERR_OUT_OF_RANGE' });
+      assert.throws(() => run({ coverage: true, functionCoverage: 101 }), { code: 'ERR_OUT_OF_RANGE' });
+
+      run({ files: [], signal: AbortSignal.abort(), coverage: true, functionCoverage: 0 });
+    });
+  });
+
+  await describe('run with coverage', skipIfNoInspector, async () => {
+    await it('should run with coverage', async () => {
+      const stream = run({ files, coverage: true });
+      stream.on('test:fail', common.mustNotCall());
+      stream.on('test:pass', common.mustCall());
+      stream.on('test:coverage', common.mustCall());
+      // eslint-disable-next-line no-unused-vars
+      for await (const _ of stream);
+    });
+
+    await it('should run with coverage and exclude by glob', async () => {
+      const stream = run({ files, coverage: true, coverageExcludeGlobs: ['test/*/test-runner/invalid-tap.js'] });
+      stream.on('test:fail', common.mustNotCall());
+      stream.on('test:pass', common.mustCall(1));
+      stream.on('test:coverage', common.mustCall(({ summary: { files } }) => {
+        const filesPaths = files.map(({ path }) => path);
+        assert.strictEqual(filesPaths.some((path) => path.includes('test-runner/invalid-tap.js')), false);
+      }));
+      // eslint-disable-next-line no-unused-vars
+      for await (const _ of stream);
+    });
+
+    await it('should run with coverage and include by glob', async () => {
+      const stream = run({ files, coverage: true, coverageIncludeGlobs: ['test/*/test-runner/invalid-tap.js'] });
+      stream.on('test:fail', common.mustNotCall());
+      stream.on('test:pass', common.mustCall(1));
+      stream.on('test:coverage', common.mustCall(({ summary: { files } }) => {
+        const filesPaths = files.map(({ path }) => path);
+        assert.strictEqual(filesPaths.some((path) => path.includes('test-runner/invalid-tap.js')), true);
+      }));
+      // eslint-disable-next-line no-unused-vars
+      for await (const _ of stream);
+    });
+
+    await it('should run with coverage and fail when below line threshold', async () => {
+      let thresholdErrorCount = 0;
+      const stream = run({ files, coverage: true, lineCoverage: 99, branchCoverage: 99, functionCoverage: 99 });
+      stream.on('test:fail', common.mustNotCall());
+      stream.on('test:pass', common.mustCall(1));
+      stream.on('test:diagnostic', common.mustCall(({ message }) => {
+        if (/Error: \d{2}\.\d{2}% (line|branch|function) coverage does not meet threshold of 99%/.test(message)) {
+          thresholdErrorCount++;
+        }
+      }));
+      // eslint-disable-next-line no-unused-vars
+      for await (const _ of stream);
+      assert.strictEqual(thresholdErrorCount, 3);
+    });
```

Comment:
```suggestion
    await it('should run with coverage and fail when below line threshold', async () => {
      let thresholdErrors = [];
      const stream = run({ files, coverage: true, lineCoverage: 99, branchCoverage: 99, functionCoverage: 99 });
      stream.on('test:fail', common.mustNotCall());
      stream.on('test:pass', common.mustCall(1));
      stream.on('test:diagnostic', common.mustCallAtLeast(({ message }) => {
        let match = message.match(/Error: \d{2}\.\d{2}% (line|branch|function) coverage does not meet threshold of 99%/);
        if (match) {
          thresholdErrors.push(match[1]);
        }
      }));
      // eslint-disable-next-line no-unused-vars
      for await (const _ of stream);
      assert.deepStrictEqual(thresholdErrors, ['line', 'branch', 'function']);
    });
```

This change will verify that `line`, `branch`, and `function` were eached called seperately.